airbyte-cdk/python/unit_tests/sources/file_based/scenarios/csv_scenarios.py

```diff
@@ -0,0 +1,225 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from airbyte_cdk.sources.file_based.exceptions import RecordParseError, SchemaInferenceError
+from unit_tests.sources.file_based.helpers import LowInferenceLimitDiscoveryPolicy
+from unit_tests.sources.file_based.scenarios._scenario_builder import TestScenarioBuilder
+
+single_csv_scenario = (
+    TestScenarioBuilder()
+    .set_name("single_csv_stream")
+    .set_config(
+        {
+            "streams": [
+                {
+                    "name": "stream1",
+                    "file_type": "csv",
+                    "globs": ["*.csv"],
+                    "validation_policy": "emit_record_on_schema_mismatch",
+                }
+            ]
+        }
+    )
+    .set_files(
+        {
+            "a.csv": {
+                "contents": [
+                    ("col1", "col2"),
+                    ("val11", "val12"),
+                    ("val21", "val22"),
+                ],
+                "last_modified": "2023-06-05T03:54:07.000Z",
+            }
+        }
+    )
+    .set_file_type("csv")
+    .set_expected_catalog(
+        {
+            "streams": [
+                {
+                    "json_schema": {"col1": "string", "col2": "string"},
+                    "name": "stream1",
+                    "supported_sync_modes": ["full_refresh"],
+                }
+            ]
+        }
+    )
+    .set_expected_records(
+        [
+            {"col1": "val11", "col2": "val12"},
```

Comment:
nit: I think we should add the filename and its timestamp to the records [like we do for the S3 source](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/stream.py#L178-L195). 

Non-blocking because I can add the filename and the cursor field as I implement incrementals